src/as2fm/jani_generator/scxml_helpers/top_level_interpreter.py

```diff
@@ -194,6 +195,38 @@ def generate_plain_scxml_models_and_timers(
     return plain_scxml_models, all_timers
 
 
+def export_plain_scxml_models(
+    generated_scxml_path: str,
+    plain_scxml_models: List[ScxmlRoot],
+    all_timers: List[RosTimer],
+    max_time: int,
+):
+    """Generate the plain SCXML files adding all compatibility entries to fit the SCXML standard."""
+    os.makedirs(generated_scxml_path, exist_ok=True)
+    models_to_export = deepcopy(plain_scxml_models)
+    global_timer_scxml = make_global_timer_scxml(all_timers, max_time)
+    if global_timer_scxml is not None:
+        models_to_export.append(global_timer_scxml)
+    # Compute the set of target automaton for each event
+    event_targets: EventsToAutomata = {}
+    for scxml_model in models_to_export:
+        for event in scxml_model.get_transition_events():
+            if event not in event_targets:
+                event_targets[event] = set()
+            event_targets[event].add(scxml_model.get_name())
+    # Add the target automaton to each event sent
+    for scxml_model in models_to_export:
+        scxml_model.add_targets_to_scxml_sends(event_targets)
```

Comment:
And this function  makes sure that we add the targets to all send tags.

Reply:
I think so, at least, this reflects what we are doing when converting the model to Jani

Reply:
I would advise adding a comment about this for future reference, but other than that it looks fine to me.